src/App.tsx

```diff
@@ -1,24 +1,55 @@
 import React from 'react';
 import './App.css';
+import GoodsList from './components/GoodsList';
 
-const goodsFromServer: string[] = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
-];
+interface State {
+  goods: string[],
+  isGoodsVisible: boolean,
+  display: string,
+}
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component<{}, State> {
+  state = {
+    goods: [
+      'Dumplings',
+      'Carrot',
+      'Eggs',
+      'Ice cream',
+      'Apple',
+      'Bread',
+      'Fish',
+      'Honey',
+      'Jam',
+      'Garlic',
+    ],
+    isGoodsVisible: false,
+    display: 'block',
+  };
+
+  getHide = () => {
+    this.setState({
+      isGoodsVisible: true,
+      display: 'none',
```

Comment:
```suggestion
      display: 'none',
```
Замість цього можна використати conditional rendering
```
{!isGoodsVisible && <button>
```